ext/mysqli/mysqli.stub.php

```diff
@@ -370,20 +349,15 @@ public function refresh(int $flags) {}
 
 class mysqli_result implements IteratorAggregate
 {
-    /** @var int|string|null */
-    public $current_field;
+    public int $current_field;
 
-    /** @var int|string|null */
-    public $field_count;
+    public int $field_count;
 
-    /** @var array|null */
-    public $lengths;
+    public ?array $lengths;
 
-    /** @var int|string|null */
-    public $num_rows;
+    public int|string $num_rows;
 
-    /** @var int|null */
-    public $type;
+    public int|null $type;
```

Comment:
This property is undocumented. I believe we should change it to return only int with an assert.

Reply:
Yes, I also think that you are right! Nice catch!